### PR TITLE
[ci] install the dunext and dunegdt wheels

### PR DESCRIPTION
**Pull request by ftalbrecht**
To debug the CI issues in #1383, this pr only updates the CI images without touching any pymor code.

### CODE REVIEW

**Review by renefritze — COMMENTED** [review]
This is a good idea :+1: 
Since the push build is already green, I fear we're looking at some weird issue that only happens if dune wheel actually got imported

**Comment by ftalbrecht** [comment]
> This is a good idea +1
> Since the push build is already green, I fear we're looking at some weird issue that only happens if dune wheel actually got imported

Great! :) Since [I have trouble running the images locally](https://github.com/pymor/docker/issues/82), I am not sure how to proceed yet. Is there an easy way to get the `ngsolve` bindings locally without these docker containers?

**Comment by ftalbrecht** [comment]
> This is a good idea +1
> Since the push build is already green, I fear we're looking at some weird issue that only happens if dune wheel actually got imported

And this would prolly happen in `core.config`? Otherwise I would see no reason for the ngsolve code to import anything touching dune...

**Comment by ftalbrecht** [comment]
Since I have trouble reproducing this locally, I will use this branch to incrementally cherry-pick stuff and see when it breakes. Since a6e281c went through, the images themselves should be fine.

**Comment by renefritze** [comment]
> Since I have trouble reproducing this locally, I will use this branch to incrementally cherry-pick stuff and see when it breakes. Since [a6e281c](https://github.com/pymor/pymor/commit/a6e281c77cb4ff524fa213e5cdbc1ed753cbbc5b) went through, the images themselves should be fine.

Why not bisect this instead? Fewer steps :wink: 

> And this would prolly happen in core.config? Otherwise I would see no reason for the ngsolve code to import anything touching dune...
Pytest discovery will import a log, including config, yes.  However running the ngslve/dune test after the discovery will ofc import still more. This all happens in the same interpreter instance btw.

**Comment by renefritze** [comment]
> s there an easy way to get the `ngsolve` bindings locally without these docker containers?

You'd have to install ngsolve in your virtualenv. That should 'only' be a matter of cloning the repo, check out correct tag and `make install`, after installing the right dev packages ...

**Comment by ftalbrecht** [comment]
So, after adding the dune check to `pymor.core.config` we already get a segfault at the end of the teste (though not at the end of the demo), similar to the error in #1383. As indicated in #1383 I can not reproduce this with `make docker_run`.

**Comment by ftalbrecht** [comment]
@sdrave Ever stumbled upon interpreter crashes involving ngsolve?

**Comment by renefritze** [comment]
Do you run the test script in the container shell after `make docker_run` or do you run `pytest` directly? 
The former starts pytest in an XvFB. That's a substantial difference for demos that want to display stuff.

**Comment by ftalbrecht** [comment]
> Do you run the test script in the container shell after `make docker_run` or do you run `pytest` directly?
> The former starts pytest in an XvFB. That's a substantial difference for demos that want to display stuff.

I ran `make docker_run` on my host and within the container executed `pytest` or ran the demos. What else should I be doing?

**Comment by renefritze** [comment]
you can run `.ci/gitlab/test_vanilla.bash` and co. to [mimic](https://docs.pymor.org/2020-2-2/developer_docs.html#stage-test) the execution of the different CI jobs. It's not exact, cause the base image for interactive `make docker_run` is the `pymor/jupyter_pyXX` one, not the `pymor/testing_pyXX` as in CI/with `make docker_test`

**Comment by ftalbrecht** [comment]
> you can run `.ci/gitlab/test_vanilla.bash` and co. to [mimic](https://docs.pymor.org/2020-2-2/developer_docs.html#stage-test) the execution of the different CI jobs. It's not exact, cause the base image for interactive `make docker_run` is the `pymor/jupyter_pyXX` one, not the `pymor/testing_pyXX` as in CI/with `make docker_test`

Wow, pyMOR's CI and docker setup is quite the beast, regarding reproducability. Nice! Nevertheless, this also fails for me currently, since `/pymor/test_results_2021.1rc0+588.g251df854.dirty.xml` and similar files are not writable. Will wait until we figure out https://github.com/pymor/docker/issues/82. Or is there a different way to `make docker_run` on my machine that is closer what you do? What is the usual scenario here?

**Comment by ftalbrecht** [comment]
So, I finally have some local solution running with `make docker_run`.
- running `cd src/pymordemos && ./thermalblock_simple.py ngsolve pod 2 5 5` passes without error
- running `pytest -sx src/pymortests/demos.py -k ngsolve` passes without error
- running `.ci/gitlab/test_vanilla.bash` passes without error but segfaults the interpreter afterwards
```python
===================================================================== 3064 passed, 44113 warnings in 1483.56s (0:24:43) ======================================================================
munmap_chunk(): invalid pointer
[b1b2a0951a64:00068] *** Process received signal ***
[b1b2a0951a64:00068] Signal: Aborted (6)
[b1b2a0951a64:00068] Signal code:  (-6)
[b1b2a0951a64:00068] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f9557313730]
[b1b2a0951a64:00068] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x10b)[0x7f9556fe87bb]
[b1b2a0951a64:00068] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x121)[0x7f9556fd3535]
[b1b2a0951a64:00068] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x79508)[0x7f955702a508]
[b1b2a0951a64:00068] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x7fc1a)[0x7f9557030c1a]
[b1b2a0951a64:00068] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x80184)[0x7f9557031184]
[b1b2a0951a64:00068] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7f9537c0de05]
[b1b2a0951a64:00068] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x39d8c)[0x7f9556fead8c]
[b1b2a0951a64:00068] [ 8] /lib/x86_64-linux-gnu/libc.so.6(+0x39eba)[0x7f9556feaeba]
[b1b2a0951a64:00068] [ 9] /usr/local/lib/libpython3.9.so.1.0(+0x1de589)[0x7f9557548589]
[b1b2a0951a64:00068] [10] /usr/local/lib/libpython3.9.so.1.0(+0x1e2a5f)[0x7f955754ca5f]
[b1b2a0951a64:00068] [11] /usr/local/lib/libpython3.9.so.1.0(PyRun_SimpleFileExFlags+0x4d2)[0x7f955754def2]
[b1b2a0951a64:00068] [12] /usr/local/lib/libpython3.9.so.1.0(Py_RunMain+0x788)[0x7f955756ad68]
[b1b2a0951a64:00068] [13] /usr/local/lib/libpython3.9.so.1.0(Py_BytesMain+0x43)[0x7f955756b1c3]
[b1b2a0951a64:00068] [14] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb)[0x7f9556fd509b]
[b1b2a0951a64:00068] [15] /usr/local/bin/python(_start+0x2a)[0x555736c0a08a]
[b1b2a0951a64:00068] *** End of error message ***
Aborted (core dumped)
```
Ideas how to build an MWE out of this, @renefritze @sdrave?

**Comment by renefritze** [comment]
Does `pytest -k gdt` crash at interpreter shutdown?
How about `xvfb -a pytest -sx src/pymortests/demos.py -k ngsolve` ?

**Comment by ftalbrecht** [comment]
> Does `pytest -k gdt` crash at interpreter shutdown?
> How about `xvfb -a pytest -sx src/pymortests/demos.py -k ngsolve` ?

There are no tests involving dune-gdt yet in this pr (just the import in `core.config`. Running `xvfb-run -a pytest -sx src/pymortests/demos.py -k ngsolve` passed without segfault, but thx for the idea!

**Comment by renefritze** [comment]
I just realized there's dolfin in the stacktrace. You should also try to tell pyMOR to not finalize MPI (ref what I said on Mattermost, I think).

**Comment by ftalbrecht** [comment]
I'll just use this as a notebook:
- `xvfb-run -a pytest -k 'not demos'` triggers the problem

**Comment by ftalbrecht** [comment]
> I just realized there's dolfin in the stacktrace. You should also try to tell pyMOR to not finalize MPI (ref what I said on Mattermost, I think).

I tried `PYMOR_MPI_FINALIZE=False xvfb-run -a pytest -k 'not demos'`, is that enough?

**Comment by renefritze** [comment]
> I tried `PYMOR_MPI_FINALIZE=False xvfb-run -a pytest -k 'not demos'`, is that enough?

Yes.

**Comment by ftalbrecht** [comment]
And we have **a winner**:

- [x] `make docker_run` and `pytest src/pymortests/core/config.py`

triggers
```python
munmap_chunk(): invalid pointer
[b1b2a0951a64:04694] *** Process received signal ***
[b1b2a0951a64:04694] Signal: Aborted (6)
[b1b2a0951a64:04694] Signal code:  (-6)
[b1b2a0951a64:04694] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f55ab591730]
[b1b2a0951a64:04694] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x10b)[0x7f55ab2667bb]
[b1b2a0951a64:04694] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x121)[0x7f55ab251535]
[b1b2a0951a64:04694] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x79508)[0x7f55ab2a8508]
[b1b2a0951a64:04694] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x7fc1a)[0x7f55ab2aec1a]
[b1b2a0951a64:04694] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x80184)[0x7f55ab2af184]
[b1b2a0951a64:04694] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7f558bfd3e05]
[b1b2a0951a64:04694] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x39d8c)[0x7f55ab268d8c]
[b1b2a0951a64:04694] [ 8] /lib/x86_64-linux-gnu/libc.so.6(+0x39eba)[0x7f55ab268eba]
[b1b2a0951a64:04694] [ 9] /usr/local/lib/libpython3.9.so.1.0(+0x1de589)[0x7f55ab7c6589]
[b1b2a0951a64:04694] [10] /usr/local/lib/libpython3.9.so.1.0(+0x1e2a5f)[0x7f55ab7caa5f]
[b1b2a0951a64:04694] [11] /usr/local/lib/libpython3.9.so.1.0(PyRun_SimpleFileExFlags+0x4d2)[0x7f55ab7cbef2]
[b1b2a0951a64:04694] [12] /usr/local/lib/libpython3.9.so.1.0(Py_RunMain+0x788)[0x7f55ab7e8d68]
[b1b2a0951a64:04694] [13] /usr/local/lib/libpython3.9.so.1.0(Py_BytesMain+0x43)[0x7f55ab7e91c3]
[b1b2a0951a64:04694] [14] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb)[0x7f55ab25309b]
[b1b2a0951a64:04694] [15] /usr/local/bin/python(_start+0x2a)[0x558384f5408a]
[b1b2a0951a64:04694] *** End of error message ***
Aborted (core dumped)
```

**Comment by ftalbrecht** [comment]
So,
- `python -c "import dune.gdt; print('dune.gdt')"` gives
  ```python
  dune.gdt
  ```
- `python -c "import dolfin; print('dolfin')"` gives
  ```python
  dolfin
  ```
- `python -c "import dolfin; print('dolfin'); import dune.xt; print('dune.xt')"` gives
  ```python
  dolfin
  dune.xt
  ```
- `python -c "import dolfin; print('dolfin'); import dune.xt; print('dune.xt'); import dune.gdt; print('dune.gdt')"` gives
  ```python
  dolfin                                                                                                                                                                                      
  dune.xt
  dune.gdt
  munmap_chunk(): invalid pointer
  [b1b2a0951a64:04803] *** Process received signal *** 
  [b1b2a0951a64:04803] Signal: Aborted (6) 
  [b1b2a0951a64:04803] Signal code:  (-6)
  [b1b2a0951a64:04803] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7fd904960730]
  [b1b2a0951a64:04803] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x10b)[0x7fd9046357bb]
  [b1b2a0951a64:04803] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x121)[0x7fd904620535]
  [b1b2a0951a64:04803] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x79508)[0x7fd904677508]
  [b1b2a0951a64:04803] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x7fc1a)[0x7fd90467dc1a]
  [b1b2a0951a64:04803] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x80184)[0x7fd90467e184]
  [b1b2a0951a64:04803] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7fd903c23e05]
  [b1b2a0951a64:04803] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x39d8c)[0x7fd904637d8c]
  [b1b2a0951a64:04803] [ 8] /lib/x86_64-linux-gnu/libc.so.6(+0x39eba)[0x7fd904637eba]
  [b1b2a0951a64:04803] [ 9] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf2)[0x7fd9046220a2]
  [b1b2a0951a64:04803] [10] python(_start+0x2a)[0x5640ee1ca08a]
  [b1b2a0951a64:04803] *** End of error message *** 
  Aborted (core dumped)
  ```
- ` python -c "import dolfin; print('dolfin'); import dune.gdt; print('dune.gdt')"` gives                                                                                                              
  ```python
  dolfin
  dune.gdt
  munmap_chunk(): invalid pointer
  [b1b2a0951a64:04826] *** Process received signal *** 
  [b1b2a0951a64:04826] Signal: Aborted (6) 
  [b1b2a0951a64:04826] Signal code:  (-6)
  [b1b2a0951a64:04826] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7fed8bbc9730]
  [b1b2a0951a64:04826] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x10b)[0x7fed8b89e7bb]
  [b1b2a0951a64:04826] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x121)[0x7fed8b889535]
  [b1b2a0951a64:04826] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x79508)[0x7fed8b8e0508]
  [b1b2a0951a64:04826] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x7fc1a)[0x7fed8b8e6c1a]
  [b1b2a0951a64:04826] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x80184)[0x7fed8b8e7184]
  [b1b2a0951a64:04826] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7fed8ae8ce05]
  [b1b2a0951a64:04826] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x39d8c)[0x7fed8b8a0d8c]
  [b1b2a0951a64:04826] [ 8] /lib/x86_64-linux-gnu/libc.so.6(+0x39eba)[0x7fed8b8a0eba]
  [b1b2a0951a64:04826] [ 9] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf2)[0x7fed8b88b0a2]
  [b1b2a0951a64:04826] [10] python(_start+0x2a)[0x55849ab3508a]
  [b1b2a0951a64:04826] *** End of error message *** 
  Aborted (core dumped)
  ```
- `python -c "import dune.gdt; print('dune.gdt'); import dolfin; print('dolfin')"` gives
  ```python
  dune.gdt
  dolfin
  ```

Thus, the **MWE** is: `python -c "import dune.gdt; import dolfin"` works, `python -c "import dolfin; import dune.gdt"` does not.

**Comment by renefritze** [comment]
Can you run the trace through `c++filt` (if it's not installed in the image, you can add it in `./binder/Dockerfile`.
Since `dune-xt` does not trigger things, I think it's unlikely related to MPI (which would be triggered through `dune-common`).
Unless the gdt import triggers more imports than dune-xt? 
Otherwise `gdt` and `dolfin` really have no relation, right?
Can you nevertheless try to check the mpi status in between importing gdt and dolfin? Something like `from mpi4py import MPI; MPI.Is_Initialized()`?

**Comment by ftalbrecht** [comment]
> Can you nevertheless try to check the mpi status in between importing gdt and dolfin? Something like `from mpi4py import MPI; MPI.Is_Initialized()`?

```python
python -c "from mpi4py import MPI; print(MPI.Is_initialized()); import dolfin; print(MPI.Is_initialized()); import dune.gdt; print(MPI.Is_initialized())"
True
True
True
munmap_chunk(): invalid pointer
[f63e8bce389e:00050] *** Process received signal ***
[f63e8bce389e:00050] Signal: Aborted (6)
[f63e8bce389e:00050] Signal code:  (-6)
[f63e8bce389e:00050] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f12ebf9c730]
[f63e8bce389e:00050] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x10b)[0x7f12ebc717bb]
[f63e8bce389e:00050] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x121)[0x7f12ebc5c535]
[f63e8bce389e:00050] [ 3] /lib/x86_64-linux-gnu/libc.so.6(+0x79508)[0x7f12ebcb3508]
[f63e8bce389e:00050] [ 4] /lib/x86_64-linux-gnu/libc.so.6(+0x7fc1a)[0x7f12ebcb9c1a]
[f63e8bce389e:00050] [ 5] /lib/x86_64-linux-gnu/libc.so.6(+0x80184)[0x7f12ebcba184]
[f63e8bce389e:00050] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7f12e8516e05]
[f63e8bce389e:00050] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x39d8c)[0x7f12ebc73d8c]
[f63e8bce389e:00050] [ 8] /lib/x86_64-linux-gnu/libc.so.6(+0x39eba)[0x7f12ebc73eba]
[f63e8bce389e:00050] [ 9] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf2)[0x7f12ebc5e0a2]
[f63e8bce389e:00050] [10] python(_start+0x2a)[0x5616bf6fe08a]
[f63e8bce389e:00050] *** End of error message ***
Aborted (core dumped)
```

**Comment by renefritze** [comment]
If https://github.com/pymor/pymor/pull/1389/commits/c91b4aa95396e0461d7cb6e4bb193aa0e87b484c helps @ftalbrecht, you can cherry-pick it over to #1383 so we can get that into the release still.

**Comment by ftalbrecht** [comment]
> Can you run the trace through `c++filt` (if it's not installed in the image, you can add it in `./binder/Dockerfile`.

The `_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev` from
```python
[f63e8bce389e:00050] [ 6] /usr/local/lib/libdolfin.so.2019.1(_ZNSt6vectorIS_IiSaIiEESaIS1_EED1Ev+0x25)[0x7f12e8516e05]
```
gives
```c++
std::vector<std::vector<int, std::allocator<int> >, std::allocator<std::vector<int, std::allocator<int> > > >::~vector()
```
which raises two questions:
- why does `dolfin` create a `std::vector<std::vector<int>>` on module import
- why does `dune-gdt` invalidate it

> Since `dune-xt` does not trigger things, I think it's unlikely related to MPI (which would be triggered through `dune-common`).
> Unless the gdt import triggers more imports than dune-xt?

No, it does not.

> Otherwise `gdt` and `dolfin` really have no relation, right?

Not that I know of, no.

> Can you nevertheless try to check the mpi status in between importing gdt and dolfin? Something like `from mpi4py import MPI; MPI.Is_Initialized()`?

```
python -c "from mpi4py import MPI; print(MPI.Is_initialized()); import dolfin; print(MPI.Is_initialized()); import dune.gdt; print(MPI.Is_initialized())"
True
True
True
```

**Comment by ftalbrecht** [comment]
If the hack works, we can work on a better solution in #1397.

**Comment by ftalbrecht** [comment]
> If [c91b4aa](https://github.com/pymor/pymor/commit/c91b4aa95396e0461d7cb6e4bb193aa0e87b484c) helps @ftalbrecht, you can cherry-pick it over to #1383 so we can get that into the release still.

Many thanks, I am obviously thinking too complicated. I applied the hack, but in the `pymor/core/config.py`, as this problem arises even without pymor and needs to be handled in the same spot where `dolfin` is imported first.

File: src/pymor/core/config.py
Comment (renefritze) [inline src/pymor/core/config.py:5]: If this needs to be in `config.py` please move it into `_get_fenics_version` if possible

Comment (ftalbrecht) [reply src/pymor/core/config.py]: also thought about that, now done

**Comment by ftalbrecht** [comment]
requires https://github.com/pymor/docker/pull/85 and a corresponding update of `dependencies.py`

**Review by sdrave — CHANGES_REQUESTED** [review]
Can the dune/dolfin hack be moved to the dolfin check? I would rather not automatically import dune whenever pymor is imported.

Also, how sure are we that importing dune has no side effects on dolfin, given that the other import order crashed?

**Comment by renefritze** [comment]
> requires [pymor/docker#85](https://github.com/pymor/docker/pull/85) and a corresponding update of `dependencies.py`

In case I don't, please restart https://zivgitlab.uni-muenster.de/pymor/pymor/-/pipelines/101726 once https://github.com/pymor/docker/pull/85 is ready

**Comment by renefritze** [comment]
> Can the dune/dolfin hack be moved to the dolfin check? I would rather not automatically import dune whenever pymor is imported.

https://github.com/pymor/pymor/pull/1389#discussion_r712796588

> 
> Also, how sure are we that importing dune has no side effects on dolfin, given that the other import order crashed?

A direct (intended) side effect? Impossible, the dune stuff is compiled w/o knowledge of dolfin. 
Indirect, as in affecting state of libs that both dolfin and gdt use? Possible, sure. See me asking about the MPI state. 
I think it's more likely some esoteric name mangling/symbol hiding issue, and/or imperfect auditwheel run.

**Comment by sdrave** [comment]
> > Also, how sure are we that importing dune has no side effects on dolfin, given that the other import order crashed?
> 
> A direct (intended) side effect? Impossible, the dune stuff is compiled w/o knowledge of dolfin.
> Indirect, as in affecting state of libs that both dolfin and gdt use? Possible, sure. See me asking about the MPI state.
> I think it's more likely some esoteric name mangling/symbol hiding issue, and/or imperfect auditwheel run.

Still, given the circumstances, someone should check before merging that numbers in the fenics demos don't suddenly change when dune has been loaded before. I'm not sure if there are any regression checks for fenics.

**Comment by ftalbrecht** [comment]
> Can the dune/dolfin hack be moved to the dolfin check? I would rather not automatically import dune whenever pymor is imported.

Do you mean as in f9fae29?

**Comment by renefritze** [comment]
> Still, given the circumstances, someone should check before merging that numbers in the fenics demos don't suddenly change when dune has been loaded before. I'm not sure if there are any regression checks for fenics.

https://github.com/pymor/pymor/blob/main/src/pymortests/demos.py#L303

**Comment by ftalbrecht** [comment]
> Can the dune/dolfin hack be moved to the dolfin check? I would rather not automatically import dune whenever pymor is imported.
> 
> Also, how sure are we that importing dune has no side effects on dolfin, given that the other import order crashed?

I also ran the `./demo_auto-adaptive_poisson.py` (we use when building the fenics images) with and without a `dune-gdt` import, and the diff is empty.

**Comment by ftalbrecht** [comment]
> > requires [pymor/docker#85](https://github.com/pymor/docker/pull/85) and a corresponding update of `dependencies.py`
> 
> In case I don't, please restart https://zivgitlab.uni-muenster.de/pymor/pymor/-/pipelines/101726 once [pymor/docker#85](https://github.com/pymor/docker/pull/85) is ready

done

**Comment by ftalbrecht** [comment]
Any idea what [this](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/708450#L179) is about, @renefritze?

**Comment by ftalbrecht** [comment]
Everything else is green, thats a start.

**Comment by renefritze** [comment]
> Any idea what [this](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/708450#L179) is about, @renefritze?

Nope. Will take a look later. 

> We have a seemingly unrelated LLVM ERROR: out of memory, which occurs only in the mpi tests.

I've restarted it once, let's see. Hopefully a temporary thing on the runner.

**Comment by ftalbrecht** [comment]
> > Any idea what [this](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/708450#L179) is about, @renefritze?
> 
> Nope. Will take a look later.
> 
> > We have a seemingly unrelated LLVM ERROR: out of memory, which occurs only in the mpi tests.
> 
> I've restarted it once, let's see. Hopefully a temporary thing on the runner.

I also restarted one of the MPI builds, did not help :/

**Comment by renefritze** [comment]
> So, everything green except the [from wheel](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/710093) part, although I simplified the version requirements.

Have a look at your mattermost messages. 

1. Rebase on main
2. Either put dune deps into the blocklist for `pymor[full]` 
3.  find and remove which if xt/gdt's transitive dependencies are responsible for needing a compiler setup to install and remove/fix those if possible. 

Given time restrictions 3. seems impossible.

**Comment by renefritze** [comment]
There's also a few linter errors blocking the merge.

**Comment by renefritze** [comment]
> > So, everything green except the [from wheel](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/710093) part, although I simplified the version requirements.
> 
> Have a look at your mattermost messages.
> 
> 1. Rebase on main
> 2. Either put dune deps into the blocklist for `pymor[full]`
> 3. OR find and remove which if xt/gdt's transitive dependencies are responsible for needing a compiler setup to install and remove/fix those if possible.
> 
> Given time restrictions 3. seems impossible.

Went ahead and did 1 and 3 hoping to see a green pipeline in the morning.

**Comment by ftalbrecht** [comment]
> > > So, everything green except the [from wheel](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/710093) part, although I simplified the version requirements.
> > 
> > 
> > Have a look at your mattermost messages.
> > 
> > 1. Rebase on main
> > 2. Either put dune deps into the blocklist for `pymor[full]`
> > 3. OR find and remove which if xt/gdt's transitive dependencies are responsible for needing a compiler setup to install and remove/fix those if possible.
> > 
> > Given time restrictions 3. seems impossible.
> 
> Went ahead and did 1 and 3 hoping to see a green pipeline in the morning.

Thx for the work and the hope! One of the `from wheel` pipelines actually succeed! Remaining errors [are associated](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/711466#L798) with [multidict building](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/711464#L665).

**Comment by renefritze** [comment]
Replaced by #1452

**Pull request closed without merging**
